### PR TITLE
implement new fuzzy search engine

### PR DESCRIPTION
**Pull request by oznakn**

### CODE REVIEW

File: maps/src/utils/filters.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export function buildSearchIndex(data: MarkerData["map_data"]) {
+  return new Fuse(data, {
+    keys: ['type', 'name', 'city', 'county'],
+    findAllMatches: true,
+  });
+}
+
+const isOneOfTypes = (item: MarkerData["map_data"][0], types: string[]) => {
+  return types.includes(item.type);
+}
```
Comment (ucanyiit) [inline maps/src/utils/filters.ts:14]: We can use a map/object here for a slightly faster filter.

File: maps/src/CustomMarker.tsx [hunk 2]
```diff
@@ -1,49 +1,50 @@
...
+          <DataItem text="Adres" value={`${item.city}${item.county ? `, ${item.county}` : ""}`} />
+          <DataItem text="Telefon" value={item.phone_number || ''} />
           <Box sx={{ mt: 2 }}>
             <Button
               sx={{ fontSize: "inherit" }}
               variant='outlined'
-              href={subitem.maps_url ? subitem.maps_url : `https://www.google.com/maps/search/?api=1&query=${subitem.latitude},${subitem.longitude}`}
+              href={item.maps_url ? item.maps_url : `https://www.google.com/maps/search/?api=1&query=${item.latitude},${item.longitude}`}
               fullWidth
             >
```
Comment (ucanyiit) [inline maps/src/CustomMarker.tsx:36]: I forgot to add `target="_blank"` here. Can you add it in this PR?

**Review by ucanyiit — APPROVED** [review]
I think it looks good. My only suggestion is to have the type text also inside the searchString so that people can see eczane's when they type "eczane".

**Comment by oznakn** [comment]
> I think it looks good. My only suggestion is to have the type text also inside the searchString so that people can see eczane's when they type "eczane".

I've already done that

**Comment by oznakn** [comment]
could you check again @ucanyiit,

also, I feel like search does not seems ok to me, it finds too many results

**Comment by oznakn** [comment]
ok I decreased the threshold, now it works fine

**Pull request merged**